src/lib/rng/chacha_rng/chacha_rng.cpp

```diff
@@ -58,20 +58,21 @@ void ChaCha_RNG::clear_state()
    m_chacha->set_key(m_hmac->final());
    }
 
-void ChaCha_RNG::generate_output(uint8_t output[], size_t output_len,
-                                 const uint8_t input[], size_t input_len)
+void ChaCha_RNG::generate_output(std::span<uint8_t> output, std::span<const uint8_t> input)
    {
-   if(input_len > 0)
+   BOTAN_ASSERT_NOMSG(!output.empty());
```

Comment:
This seems wrong - `randomize(nullptr, 0)` should succeed and do nothing. (There should be a test for this)

Reply:
It looks like there are tests along these lines for `System_RNG` and `Processor_RNG` but not the stateful implementations

Reply:
Added a test covering this in a379e17340ce26a6f71fd59e3b5ec352d2cce152

Reply:
The `generate_output()` method is the internal virtual entrypoint of `Stateful_RNG`. The implementation in `Stateful_RNG::fill_bytes_with_input()` is marked as `final` and won't call into `generate_output()` if the given output sink is empty. I.e. this is just a safety net.